src/lib.rs

```diff
@@ -235,54 +235,61 @@ fn append(result: &mut Vec<u8>, to_append: &[u8], bytes_on_line: &mut usize, bac
 /// ```
 ///     use quoted_printable::encode;
 ///     let encoded = encode("hello, \u{20ac} zone!".as_bytes());
-///     assert_eq!("hello, =E2=82=AC zone!", String::from_utf8(encoded).unwrap());
+///     assert_eq!("hello, =E2=82=AC zone!", encoded);
 /// ```
-pub fn encode(input: &[u8]) -> Vec<u8> {
+#[inline(always)]
+pub fn encode<R: AsRef<[u8]>>(input: R) -> String {
+    _encode(input.as_ref())
+}
```

Comment:
This additional delegation is a little optimization wrt. monomorphization and used in this way in many placed in `std`.
(we basically inline the `.as_ref()` call into the calling code making 100% sure there is only
one instance of `encode` in the binary)